### DISCUSSION TITLE
Enclosed as replacement for Wetransfer?

### DISCUSSION THREAD

**Discussion by Martin511** [CorentinTh/enclosed#364] [Q&A] [answered]
I have Enclosed installed on my NAS and I am very happy with it.
Today I searched for a self hosted solution as an alternative for Wetransfer.
Then I asked myself: Can Enclosed be that? What is the max. file size? Can I send files of a couple of GB?
Will there be a feature in the future to enter the recipients email address so he gets noticed automatiaclly?
What do you think?

**Comment by Zoobdude** [comment, marked as answer]
Hey @Martin511,

Enclosed could probably be a Wetransfer replacement. The maximum file size on self hosted instances can be set via the `NOTES_MAX_ENCRYPTED_PAYLOAD_LENGTH` ENV variable ([full list can be found here](https://docs.enclosed.cc/self-hosting/configuration)).

I've sent pretty big files using it successfully, the main issue I ran into is my reverse proxy timing out the connection as the download takes quite a while. This however, can easily be fixed with a configuration change.

As for email. This is unlikely to be implemented as it would involve the server having knowledge of the note URL and therefore encryption key, defeating the purpose of the e2e encryption. I described this in another issue regarding making plain file uploads via an API. https://github.com/CorentinTh/enclosed/issues/352#issuecomment-2499913067

**Comment by Martin511** [comment]
Thank you.
Meanwhile I came to the point that I will try to use Thunderbird SEND to send bigger files as I use Thunderbird anyway.
This works good so far, I will try to install my own SEND instance.
